dali/operators/generic/resize/tensor_resize.h

```diff
@@ -0,0 +1,207 @@
+// Copyright (c) 2022-2023, NVIDIA CORPORATION & AFFILIATES. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef DALI_OPERATORS_GENERIC_RESIZE_TENSOR_RESIZE_H_
+#define DALI_OPERATORS_GENERIC_RESIZE_TENSOR_RESIZE_H_
+
+#include <cassert>
+#include <memory>
+#include <string>
+#include <utility>
+#include <vector>
+#include "dali/core/expand_dims.h"
+#include "dali/operators/image/resize/resize_base.h"
+#include "dali/operators/image/resize/tensor_resize_attr.h"
+#include "dali/pipeline/operator/common.h"
+#include "dali/pipeline/operator/operator.h"
+
+namespace dali {
+namespace tensor_resize {
+
+
+template <typename Backend>
+class TensorResize : public Operator<Backend>
+                   , protected ResizeBase<Backend> {
+ public:
+  explicit TensorResize(const OpSpec &spec);
+
+ protected:
+  int NumSpatialDims() const { return spatial_ndim_; }
+  int FirstSpatialDim() const { return first_spatial_dim_; }
+
+  bool CanInferOutputs() const override { return true; }
+
+  bool SetupImpl(std::vector<OutputDesc> &output_desc, const Workspace &ws) override;
+
+  void RunImpl(Workspace &ws) override;
+
+  void TrimSpatialDims(const TensorListShape<> &input_shape, int min_ndim) {
+    int nsamples = input_shape.num_samples();
+    auto unchanged_dim = [&](int d) {
+      for (int i = 0; i < nsamples; i++) {
+        int64_t extent = input_shape.tensor_shape_span(i)[d];
+        if (static_cast<int64_t>(resize_params_[i].dst_size[d]) != extent ||
+            static_cast<int64_t>(resize_params_[i].src_lo[d]) != 0 ||
+            static_cast<int64_t>(resize_params_[i].src_hi[d]) != extent) {
+          return false;
+        }
+      }
+      return true;
+    };
+
+    // at least min_ndim should remain
+    int can_trim_n = std::max(0, spatial_ndim_ - min_ndim);
```

Comment:
Nitpick:
`can_trim_n` reads like a boolean (can I trim `n` dimensions?)
```suggestion
    int ndim_to_trim = std::max(0, spatial_ndim_ - min_ndim);
```